docker/gcp_pudl_etl.sh

```diff
@@ -97,6 +99,17 @@ function notify_slack() {
     send_slack_msg "$message"
 }
 
+if [ "$GITHUB_ACTION_TRIGGER" = "workflow_dispatch" ]; then
+    echo "Deployed via workflow_dispatch, testing git authentication!"
+    git config user.email "pudl@catalyst.coop"
+    git config user.name "pudlbot"
+    git remote set-url origin "https://pudlbot:$PUDL_BOT_PAT@github.com/catalyst-cooperative/pudl.git"
+    git config -l
+    git tag -a -m "Nightly test tag" nightly-tag-test
+    git push origin nightly-tag-test
+    shutdown_vm
+fi
+
```

Comment:
This is temporary -- the idea was to test whether this authentication and tag push worked without running a bunch of unnecessary stuff but... it never seems to have run and I didn't get any error output in the container logs so I don't know what is going on here.

Reply:
I do like the idea of having this sort of short circuit test thing in the script for now, I think that will make testing this way less painful.

I see some logs corresponding to this in the [Log Explorer...](https://console.cloud.google.com/logs/query;query=resource.type%3D%22gce_instance%22%20resource.labels.zone%3D%22us-west1-a%22%20%2528resource.labels.instance_id%3D%222074811782323017117%22%20OR%20protoPayload.resourceName%3D%22projects%2Fcatalyst-cooperative-pudl%2Fzones%2Fus-west1-a%2Finstances%2Fpudl-deployment-tag%22%2529%0Aseverity%3E%3DDEFAULT%0AjsonPayload.%22cos.googleapis.com%2Fcontainer_name%22%3D%22klt-pudl-deployment-tag-luui%22;cursorTimestamp=2023-12-20T05:33:35.290711021Z;startTime=2023-12-20T05:11:00.000Z;endTime=2023-12-20T06:45:00.000Z?referrer=search&project=catalyst-cooperative-pudl)

That shows a couple things:

1. Hah, it logs out some sensitive information. Oops.
2. It does appear to run, just failing authentication. Are you able to use that `pudlbot:<TOKEN>` method to authenticate locally? I can give it a shot now.

Reply:
I was able to `git remote add origin-pudlbot https://pudlbot:<TOKEN>@github.com/...` and then push a new branch with `git push -u origin-pudlbot <branch-name>`.

Reply:
The `git config -l` output in the logs does show an extra config value that I don't have locally: `http.https://github.com/.extraheader=AUTHORIZATION: basic <token>` - that seems suspicious.

Reply:
I think that basic auth token probably gets copied in as part of the docker build process, since we're copying in the whole `.git` directory which was checked out by the `checkout` GitHub action.

I think we should avoid baking in any sensitive tokens to the Docker image 😉 but we could potentially try removing that `http.https://github.com....` config value and see if it chooses the right auth.